modules/nosql/manager_leveldb.go

```diff
@@ -48,12 +56,28 @@ func (m *Manager) GetLevelDB(connection string) (*leveldb.DB, error) {
 
 		return db.db, nil
 	}
-	uri := ToLevelDBURI(connection)
+
+	db, ok = m.LevelDBConnections[uri.String()]
+	if ok {
+		db.count++
+
+		return db.db, nil
+	}
+
+	// if there is already a connection to this leveldb reuse that
+	// NOTE: if there differing options then only the first leveldb connection will be used
+	db, ok = m.LevelDBConnections[dataDir]
+	if ok {
+		db.count++
+		log.Warn("Duplicate connnection to level db: %s with different connection strings. Initial connection: %s. This connection: %s", dataDir, db.name[0], connection)
+		db.name = append(db.name, connection)
+		m.LevelDBConnections[connection] = db
```

Comment:
```suggestion
```

If we run into this `if`, then neither `connection` or `uri` exists in the `LevelDBConnections`.

But here the code only append the `connection` to the name list, but no `uri`, is it by purpose?

In my opinion:
* Either we keep consistent that we append both non-existing `connection` and `uri` into the name list
* Or append nothing (remove these 2 lines), because the logic is sill correct, no performance degrade, and more simple.

Since the old code works, this comment is not a blocker.

Reply:
It prevents the repeated logging of the same message

Reply:
But it's not complete. 

Current code It only prevents repeated logging for the new `connection`, but not for the new `uri` with the same directory. (that the case 1)

And I do not think there will be repeated loggings, the queues call `GetLevelDB` for `open the db`, there seems no repeated calls (so that why case 2 is also fine because there seems to be no repeated log messages)

Reply:
Yes it absolutely does prevent repeated logging of the same message. The log message doesn't contain the inferred URI. The point is to tell the server user which strings in their configuration are incorrect.

There are multiple queues. Depending on the order of initialisation and state of the incorrect configuration without this you face repeatedly seeing the same warning message.